### PR TITLE
Updated UI of client profile and edit client profile

### PR DESCRIPTION
**Pull request by giteshk11**
# Issue Being Addressed
#227 

Put the issue(s) # that your PR is addressing here. e.g. #15, #29

# Type of PR

Put an `x` in the brackets for the type(s) of PR this is. You may tick more than one.

[ ] Bug Fix
[x] Refactor
[x] New Feature
[ ] Update to Existing Feature
[ ] Other (state below)

# Screenshots/casts

If applicable, include any relevant screenshot or screencasts of your changes.
Screen Shot 2020-08-27 at 12 12 33
Screen Shot 2020-08-27 at 12 12 40

# Additional Context

Provide any additional notes that you think would help effective review of your code.

### CODE REVIEW

**Review by paulchrisluke — APPROVED** [review]
ux looks good, @bamboochow is updated my account to be similar, we can use the same thing there.

File: src/layouts/WithTitleBarLayout.vue [hunk 1]
```diff
@@ -1,17 +1,47 @@
...
         <IconBack class="text-white mr-2" />
       </a>
-      <h1 class="tg-h2-mobile text-opacity-high text-white ml-2">
+      <h1
+        class="tg-h2-mobile text-opacity-high flex-grow text-left text-white ml-2"
+      >
         {{ currentTitle }}
       </h1>
+      <IconOverflowMenu
+        v-if="$route.name === 'ClientEdit'"
```
Comment (mort3za) [inline src/layouts/WithTitleBarLayout.vue:16]: Having this 'more' icon here is ok but the action should be agnostic to routes, this is a layout and can be used in a hundred pages.  
Pass the action as an event like what we have currently for back button. Event data can be something like: `{title: 'PMU Sign', type: 'routeChange', route: {name: 'PMUSign', params: {paramX:1}}}}` or `{title: 'Log out', type: 'storeAction', action: 'auth/logout'}`

File: src/views/EditClientInfo.vue [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+      </p>
+    </MaterialInput>
+    <MaterialInput
+      :margin="null"
+      v-model="$v.client.email.$model"
+      label="Email"
+      :error="$v.client.email.$error"
+      :attrs="{ readonly: true }"
+    >
+      <p v-if="!$v.client.email.required" class="text-error text-xs">
```
Comment (mort3za) [inline src/views/EditClientInfo.vue:49]: No need to use text-error and text-xs here. We have required error classes in MaterialInput component.

File: src/views/EditClientInfo.vue [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+        });
+      });
+    },
+
+    archive() {
+      this.archiveClient({
+        tenantSlug: this.tenantSlug,
+        clientId: this.clientId
+      }).then(async () => {
+        this.isDeleteModalOpen = false;
```
Comment (mort3za) [inline src/views/EditClientInfo.vue:178]: Use `showOverlayAndRedirect` from helpers.js instead.

**Review by mort3za — CHANGES_REQUESTED** [review]
Everything else looks good!

File: src/layouts/WithTitleBarLayout.vue (hunk 1 above)
Comment (giteshk11) [reply src/layouts/WithTitleBarLayout.vue:16]: can you explain more ?

File: src/views/EditClientInfo.vue (hunk 3 above)
Comment (giteshk11) [reply src/views/EditClientInfo.vue:178]: like this `await showOverlayAndRedirect();` ?

Comment (mort3za) [reply src/views/EditClientInfo.vue:178]: There are some usage examples in the repo, e.g. https://github.com/whynotearth/BrowTricks/blob/staging/src/views/ClientNotes.vue#L121

Comment (mort3za) [reply src/views/EditClientInfo.vue:178]: And https://github.com/whynotearth/BrowTricks/blob/staging/src/views/ClientNotifications.vue#L105

File: src/layouts/WithTitleBarLayout.vue (hunk 1 above)
Comment (mort3za) [reply src/layouts/WithTitleBarLayout.vue:16]: Well, layout is a high level component, high level components should not depend on low levels.  
In a layout file we don't know anything from the inner components (like client edit), we just provide a way for client edit to update layout (adding menu items to top right).
This updating way is listening for new menu items (we don't know the menu items, we just get them and put them on it's place).

Comment (mort3za) [reply src/layouts/WithTitleBarLayout.vue:16]: If it was confusing, don't worry, just add a TODO: REFACTOR and leave it to me.

Comment (giteshk11) [reply src/layouts/WithTitleBarLayout.vue]: I tired to solve it.. let me if this approach is fine

Comment (mort3za) [reply src/layouts/WithTitleBarLayout.vue]: Perfect.

**Pull request merged**